### PR TITLE
Ports Condiment Sprites from TG. Ports Mayonnaise, Peanut Butter, Olives, and Olive Oil from TG

### PR DESCRIPTION
**Pull request by Mitchs98**
## What Does This PR Do
Freshens up our old condiment sprites from TG, as well as the serving tray(it was also in the same PR so why not). Also minorly refactors condiment bottle code, does not change any functionality.

https://github.com/tgstation/tgstation/pull/74890 - Salt and Pepper Shaker by CoiledLamb.
https://github.com/tgstation/tgstation/pull/75715 - Condiments by MTandi.

Adds updated empty condiment bottle, mixed condiment bottle, and wasabi tube by ATP Engineer

Mayonnaise is created with 5 egg, 1 salt, and 5 Olive Oil resulting in 10 Mayonnaise.

Olive Oil is created with 5 Olive Paste and 5 Water resulting in 10 Olive Oil.

Olive Paste is made by grinding up Olives.

Peanut Butter is created by grinding up Peanuts.

## Why It's Good For The Game
Our current condiment sprites are, quite honestly, **very bad**. This PR aims to improve that while also adding a few new condiments that will be used in future recipes I intend on porting from TG. Essentially lays the groundwork and adds a new condiment, two if you count peanut butter. Also makes our serving tray look more like a serving tray and less like a baking tray.

Cleaver sprite is also updated to be more in line with our other sprites, ported from TG.

## Images of changes
dreammaker_ZdlDtEqN3d

dreammaker_9p7wkcbeeY

newthings

## Testing
1. Spawned in each and every condiment bottle.
2. Filled condiment bottles with their respective condiments.
3. Grew an Olive tree, cried at the offsetting, and properly offset it by the needed amount.
4. Harvested Olives.
5. Ground Olives and Peanut Butter.
6. Mixed Olive Oil.
7. Mixed Mayonnaise.
8. Ate the Mayonnaise.
9. Drank the excess Olive Oil.

## Changelog
:cl: Mitchs98, ATP Engineer, CoiledLamb, MTandi
add: Added Mayonnaise, created by mixing 5 egg, 1 salt, and 5 Olive Oil resulting in 10 units of Mayonnaise.
add: Adds Olive Oil and Olive Paste. Olive Oil is made from 5 Olive Paste and 5 Water, resulting in 10 units. Olive Paste is acquired by grinding olives.
add: Peanut Butter, Acquired by grinding Peanuts.
add: Olives, growable in botany. Mutates into nothing, provides Olive Paste when ground.
imageadd: Most Condiments have been given new sprites ported from TG.
imageadd: Ported updated Tray sprite from TG.
imageadd: Ports TG's Cleaver sprite.
imageadd: Condiment bottle, mixed condiment bottle, and Wasabi bottle by ATP Engineer.
/:cl:

### CODE REVIEW

**Comment by SynthTwo** [comment]
YES LETS GOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOO

File: code/modules/food_and_drinks/food/condiment.dm [hunk 1]
```diff
@@ -24,9 +24,14 @@
 	"frostoil" = list("coldsauce", "coldsauce bottle", "Leaves the tongue numb in it's passage"),
 	"sodiumchloride" = list("saltshakersmall", "salt shaker", "Salt. From space oceans, presumably"),
 	"blackpepper" = list("peppermillsmall", "pepper mill", "Often used to flavor food or make people sneeze"),
-	"cornoil" = list("oliveoil", "corn oil bottle", "A delicious oil used in cooking. Made from corn"),
+	"cornoil" = list("cornoil", "corn oil bottle", "A delicious oil used in cooking. Made from corn"),
```
Comment (Henri215) [inline code/modules/food_and_drinks/food/condiment.dm:27]:
```suggestion
	"cornoil" = list("cornoil", "corn oil bottle", "A delicious oil used in cooking. Made from corn."),
```
While we are here...

File: code/modules/food_and_drinks/food/condiment.dm [hunk 2]
```diff
@@ -218,7 +258,15 @@
...
+	possible_states = list(
+	"ketchup" = list("condi_ketchup", "Ketchup", "You feel more American already."),
+	"capsaicin" = list("condi_hotsauce", "Hotsauce", "You can almost TASTE the stomach ulcers now!"),
+	"soysauce" = list("condi_soysauce", "Soy Sauce", "A salty soy-based flavoring"),
+	"frostoil" = list("condi_frostoil", "Coldsauce", "Leaves the tongue numb in it's passage"),
+	"sodiumchloride" = list("condi_salt", "Salt Shaker", "Salt. From space oceans, presumably"),
+	"blackpepper" = list("condi_pepper", "Pepper Mill", "Often used to flavor food or make people sneeze"),
+	"cornoil" = list("condi_cornoil", "Corn Oil", "A delicious oil used in cooking. Made from corn"),
```
Comment (Henri215) [inline code/modules/food_and_drinks/food/condiment.dm:268]:
```suggestion
	"soysauce" = list("condi_soysauce", "Soy Sauce", "A salty soy-based flavoring."),
	"frostoil" = list("condi_frostoil", "Coldsauce", "Leaves the tongue numb in it's passage."),
	"sodiumchloride" = list("condi_salt", "Salt Shaker", "Salt. From space oceans, presumably."),
	"blackpepper" = list("condi_pepper", "Pepper Mill", "Often used to flavor food or make people sneeze."),
	"cornoil" = list("condi_cornoil", "Corn Oil", "A delicious oil used in cooking. Made from corn."),
```

File: code/modules/hydroponics/grown/olive.dm [hunk 3]
```diff
@@ -0,0 +1,24 @@
+// Olives!
+/obj/item/seeds/olive
+	name = "pack of olive seads"
+	desc = "These seeds grow into olive trees"
```
Comment (Henri215) [inline code/modules/hydroponics/grown/olive.dm:4]:
```suggestion
	desc = "These seeds grow into olive trees."
```

**Comment by Mitchs98** [comment]
Appended periods to stuff Henri mentioned, also added spawnable bottles of the rest of the condiments because why not? Also finished pathing the honey and etc. on the condiments list because I forgor to do that surprised it compiled properly the first time and worked fine.

> It never dawned on me till now...but we dont have mustard

I honestly don't know *any* server that has mustard. I know TG doesn't, I was going to port it from there but it sadly doesn't exist.

**Comment by DT2399** [comment]
YES!!

**Comment by Mitchs98** [comment]
Ported TGs updated cleaver, looks much shinier and nicer than ours. Also removed the bloody cleaver that I didn't see used *anywhere*. PR Desc updated shortly.

**Comment by Migratingcocofruit** [comment]
>A small cylindrical salty fruit
 
I know it's a port but considering this description is would make sense for it to have a small amount of salt as a reagent.

**Comment by Probablyjay19** [comment]
Can I use mayonnaise as an instrument?

**Review by Henri215 — CHANGES_REQUESTED** [review]
image
Remove that duplicate sprite

**Comment by Mitchs98** [comment]
> image Remove that duplicate sprite

I did. THat's just a reference picture :p

**Comment by Mitchs98** [comment]
> Can I use mayonnaise as an instrument?

Perhaps soon.

**Comment by Mitchs98** [comment]
> > A small cylindrical salty fruit
> 
> I know it's a port but considering this description is would make sense for it to have a small amount of salt as a reagent.

Missed this earlier. I'll consider doing that, wanted to try and avoid grinder bloat when trying to make oliveoil...but at the same time it would streamline making oliveoil a bit.

I think I'll push an update for this later tonight.

**Comment by Migratingcocofruit** [comment]
> > > A small cylindrical salty fruit
> > 
> > 
> > I know it's a port but considering this description is would make sense for it to have a small amount of salt as a reagent.
> 
> Missed this earlier. I'll consider doing that, wanted to try and avoid grinder bloat when trying to make oliveoil...but at the same time it would streamline making oliveoil a bit.
> 
> I think I'll push an update for this later tonight.
> 
> Olives will now have a bit of salt in them.

The grinder has both grind and juice options. IIRC Grind gives the reagents and juice usually gives a reagent that is dependent on the plant. So you can make it give the mahsed olives on juice and the reagents on grind

**Comment by Mitchs98** [comment]
> > > > A small cylindrical salty fruit
> > > 
> > > 
> > > I know it's a port but considering this description is would make sense for it to have a small amount of salt as a reagent.
> > 
> > 
> > Missed this earlier. I'll consider doing that, wanted to try and avoid grinder bloat when trying to make oliveoil...but at the same time it would streamline making oliveoil a bit.
> > I think I'll push an update for this later tonight.
> > Olives will now have a bit of salt in them.
> 
> The grinder has both grind and juice options. IIRC Grind gives the reagents and juice usually gives a reagent that is dependent on the plant. So you can make it give the mahsed olives on juice and the reagents on grind

A juicing setting typically wouldn't produce paste, it would mulchify the olive. Going to keep it as is, really.

**Review by S34NW — APPROVED** [review]

**Pull request merged**